marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -203,8 +203,10 @@ const MappingCard: React.FC<Props> = (props) => {
                 let parsedDoc: any;
                 if(typeof(srcDocResp.data) === 'string'){
                     parsedDoc = getParsedXMLDoc(srcDocResp);
+                    setSourceFormat('xml');
```

Comment:
This is fine, but the backend should really provide all of this data - for 5.4, we've gotta do a better job of shifting more of the burden to the backend so that life is as easy as possible for the UI.

Reply:
Agreed, it would be better to have the backend tell the front end the source format explicitly.